embassy/src/util/mutex.rs

```diff
@@ -78,3 +105,30 @@ pub fn in_thread_mode() -> bool {
     return cortex_m::peripheral::SCB::vect_active()
         == cortex_m::peripheral::scb::VectActive::ThreadMode;
 }
+
+/// A "mutex" that does nothing and cannot be shared between threads.
+pub struct NoopMutex<T> {
+    inner: UnsafeCell<T>,
```

Comment:
this could be simply `inner: T`

Reply:
Nice one! Commit f5199cbbf3df9146e13baa06d5b63863c8794e15 attends to this.